src/Compilers/CSharp/Portable/Emitter/Model/PEAssemblyBuilder.cs

```diff
@@ -264,6 +266,19 @@ internal override SynthesizedAttributeData SynthesizeScopedRefAttribute(WellKnow
             return base.SynthesizeScopedRefAttribute(member);
         }
 
+        internal override SynthesizedAttributeData SynthesizeRefSafetyRulesAttribute(ImmutableArray<TypedConstant> arguments)
```

Comment:
We will probably need to synthesize this attribute in some cases, even if it's included in .NET 7 runtime, because the attribute is needed when upgrading to C#11 and recompiling a module that references .NET 6.

Reply:
> We will probably need to synthesize this attribute in some cases, even if it's included in .NET 7 runtime, because the attribute is needed when upgrading to C#11 and recompiling a module that references .NET 6.

The entire point I was making is that this is not a supported scenario. Users can define the attribute for these cases. Did we come to a decision on whether we should be synthesizing this attribute?

Reply:
At this point we need to be synthesizing the attribute. It's very late in the `net7.0` game. Not synthesizing the attribute would put RC2 at risk and cause a lot of breaking change headaches when adopting. It's just too late to not do this

Reply:
Alright, if we must.